source/transactions/transactions.rst

```diff
@@ -744,24 +753,70 @@ format:
 
     { ok : 1, writeConcernError: {code: <Number>, errmsg : "..."} }
 
-Mongos Pinning for Sharded Transactions
----------------------------------------
+Sharded Transactions
+--------------------
+
+MongoDB 4.2 (maxWireVersion 8) introduces support for sharded transactions.
+Sharded transactions support all of the same features as single replica set
+transaction but introduce two new driver concepts: mongos pinning and the
+``recoveryToken`` field.
 
-The server supports sharded transactions starting in MongoDB 4.2 (
-maxWireVersion 8). The server requires that drivers MUST send all commands for
-a single transaction to the same mongos.
+Mongos Pinning
+~~~~~~~~~~~~~~
+
+Drivers MUST send all commands for a single transaction to the same mongos
+(excluding retries of commitTransaction and abortTransaction).
 
 After the driver selects a mongos for the first command within a transaction,
 the driver MUST pin the ClientSession to the selected mongos. Drivers MUST
-send all commands that are part of the same transaction, including
-commitTransaction and abortTransaction and any retries thereof, to the
-same mongos.
+send all subsequent commands that are part of the same transaction (excluding
+retries of commitTransaction and abortTransaction) to the same mongos.
+
+When to unpin
+^^^^^^^^^^^^^
+
+Drivers MUST unpin a ClientSession when a command within a transaction
+fails with a TransientTransactionError. Transient errors indicate that the
+transaction in question has already been aborted or that the pinned mongos is
+down/unavailable. Unpinning the session ensures that a subsequent
+abortTransaction does not block waiting on a server that is unreachable.
+
+Additionally, Drivers MUST unpin a ClientSession after running any
+commitTransaction or abortTransaction attempt regardless if that attempt
+succeeded or failed. After the initial commit or abort attempt, any mongos can
+satisfy a subsequent retry. Note, when the initial attempt fails with a
+retryable error, the automatic retry attempt MUST perform normal
+server selection to select an available mongos.
 
 Starting a new transaction on a pinned ClientSession MUST unpin the
 session. Additionally, any non-transaction operation using a pinned
 ClientSession MUST unpin the session and the operation MUST perform normal
 server selection.
 
+recoveryToken field
+~~~~~~~~~~~~~~~~~~~
+
+The ``recoveryToken`` field enables the driver to recover the outcome of a
+sharded transaction on a new (or restarted) mongos. [#]_
```

Comment:
not any ops, but on commitTransaction commands that had an ambiguous result: like network error.

Reply:
Can you expand on this? I'm not sure if you're asking for changes or clarification.

Reply:
The sentence sounds like recoveryToken can be used to recover the outcome of any ops in a sharded transaction. The token is only meant to recover the outcome of a commitTransaction command.

Reply:
Can you suggest a different phrasing? I'm not sure how to say "recover the outcome of a sharded transaction" more clearly.

Does the footnote below clarify this sufficiently? `In 4.2, a new mongos waits for the *outcome* of the transaction but...`

Reply:
Perhaps just say "enables the driver to recover the outcome of a commitTransaction.

Reply:
I think the current wording is fine because the sharded txn design doc uses the same language, see "Best effort to **recover a transaction’s outcome** via secondary router"

I just reordered it slightly to match.